src/common/snippets/include/snippets/utils/tokenization_utils.hpp

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A file contains tokenization related utilities.
+ * @file tokenization_utils.hpp
+ */
+#pragma once
+
+#include "snippets/op/subgraph.hpp"
+#include "snippets/pass/tokenization.hpp"
+
+namespace ov {
+namespace snippets {
+namespace utils {
+/**
+ * @brief Tokenizes a node into Subgraph. 2 options are possible (depending on config's values and internal logic)L
+ *        1. The node is wrapped in a trivial Subgraph which contains only this node
+ *        2. The node is fused in parent's Subgraphs
+ * @param node node which should be tokenized
+ * @param config tokenization config which regulates 
+ * @return whether the node was tokenized or not
+ */
+bool tokenize_node(const std::shared_ptr<ov::Node>& node, const ov::snippets::pass::SnippetsTokenization::Config& config);
```

Comment:
Probably we need to create the static public method in the pass "TokenizeSnippets" instead of new file creation with one function. What do you think?

Reply:
I decided to move this in a separate file because
1. This is a common helper which can be reused in numerous tokenization passes in the future
2. The helper is large, so it's a bit easier to have it in a separate file from readability perspective (At least that's what it seems to me 😄 )
3. This file may be extended with `bool tokenize_nodes(const ov::NodeVector& nodes)` helper which tokenizes a nodes sequence. Now this is a part of `TokenizeMHASnippets` pass, but this code can be easily extracted and reused (e.g. for MLP tokenization)

Reply:
I think we should go even further and create a dedicated class for tokenization (`SnippetsNodeTokenizer` or smth). This class should have 3 methods: private  `bool check_necessary(node)` public virtual `bool check_sufficient(node)` and public `bool fuse(node)`.
1) `check_necessary` will perform all the fundamental plugin-independent checks (like cyclic dependencies, num_results_children etc) and can potentially also gather some aggregated info (internal/external inputs/body parameters etc). If possible, we should separate the check from aggregated info gathering (another method `analyze`).
2) `check_sufficient` will be overrided by plugin and will contain backend-specific checks possibly partially based on the gathered aggregated info (hidden_data_count, unique_biffer_count) etc
3) Finally, `fuse` will call `if(check_necessary() && check_sufficient())` and create subgraph based on aggregated info.
So the idea is that the plugin creates its own instance of this `SnippetsNodeTokenizer` and passes it to all the tokenization transformations. And the transformations can use this class to do all the dirty work, plus they can impose their own limitations of course. For example, the eltwise tokenization pass will additionally check op types and precisions.

@a-sidorova, @v-Golubev what do you think? I'm nit saying we should refactor it know, but let's try to outline a more convenient and scalable design.

Reply:
@IvanNovoselov your idea looks interesting, but I need some time to think if the proposed architecture is enough. Can we allocate a time slot in the next command meeting to discuss it?

Reply:
Seems like we really need to discuss how the current logic of the tokenization can be refactored and improved.